cuebot/src/main/java/com/imageworks/spcue/dispatcher/HostReportHandler.java

```diff
@@ -309,13 +327,60 @@ else if (!dispatchSupport.isCueBookable(host)) {
      * updated with a boot report.  If the state is Repair, then state is
      * never updated via RQD.
      *
+     *
+     * Prevent cue frames from booking on hosts with full MCP directories.
+     *
+     * Change host state to REPAIR or UP according the amount of free space
+     * in the /mcp directory:
+     * - Set the host state to REPAIR, when the amount of free space in the
+     * /mcp directory is less than the minimum required. Add a comment with
+     * subject: "Action required. Host status = Repair. Reason: Full /mcp directory".
+     * - Set the host state to UP, when the amount of free space in the /mcp directory
+     * is greater or equals to the minimum required and the host has a comment with
+     * subject: "Action required. Host status = Repair. Reason: Full /mcp directory".
+     *
      * @param host
      * @param reportState
      * @param isBoot
+     * @param freeMcp
+     * @param minBookableFreeMCP: The minimum amount of free space in the MCP directory to book a host
      */
-    private void changeHardwareState(DispatchHost host,
-            HardwareState reportState, boolean isBoot) {
+    private void changeHardwareState(DispatchHost host, HardwareState reportState, boolean isBoot, long freeMcp,
+                                     Long minBookableFreeMCP) {
```

Comment:
nit: can we call `env.getRequiredProperty("dispatcher.min_bookable_free_mcp_kb", Long.class)` to fetch this value instead of passing it as a function parameter?

Fetching properties should be cheap.

Reply:
Hi Brian.

I updated the code following you recommendation. Now the solution is calling ```env.getRequiredProperty("dispatcher.min_bookable_free_mcp_kb", Long.class)``` twice. Inside the ```changeHardwareState()``` method and inside ```handleHostReport()``` method.

Please note that currently the only place the ```changeHardwareState()``` method is called is in the ```handleHostReport()``` method. My previous implementation did only one call of ```env.getRequiredProperty("dispatcher.min_bookable_free_mcp_kb", Long.class)``` because the minBookableFreeMCP variable was used later on in the ```handleHostReport()``` method to log the message of minimum amount of free space.

- Use of ```minBookableFreeMCP``` latter in the ```handleHostReport()``` method:
```
if (minBookableFreeMCP != -1 && report.getHost().getFreeMcp() < minBookableFreeMCP) {
                msg = String.format("%s doens't have enough free space in the /mcp directory, %dMB needs %dMB",
                        host.name, (report.getHost().getFreeMcp()/1024),  (minBookableFreeMCP/1024));
            }
```

However, I understand your point of view that the changeHardwareState method should know the min_bookable_free_mcp_kb value and that etching properties should be cheap.